src/dlgTriggerEditor.cpp

```diff
@@ -7324,7 +7324,9 @@ void dlgTriggerEditor::slot_changeEditorTextOptions( QTextOption::Flags state )
 void dlgTriggerEditor::clearDocument(edbee::TextEditorWidget* ew, const QString& initialText) {
 
     // This will cause a crash on the next redoClear because of an unregistered view and a >0 redo count
-    mpSourceEditorEdbeeDocument = newTextDocument();
+
+    mpSourceEditorEdbeeDocument = new edbee::CharTextDocument();
+    mpSourceEditorEdbeeDocument->setLanguageGrammar(edbee::Edbee::instance()->grammarManager()->detectGrammarWithFilename(QLatin1Literal("Buck.lua")));
```

Comment:
The configuration carries over from the widget, but not the filetype for the lexer, so that needs to be set for the new document.